### PR TITLE
MAINT: fix unhashable instance and potential exception identified by LGTM.

### PR DESCRIPTION
**Pull request by sistaseetaram**
Apropos #19077.

### CODE REVIEW

File: numpy/linalg/lapack_lite/clapack_scrub.py [hunk 1]
```diff
@@ -299,6 +299,6 @@ def scrubSource(source, nsteps=None, verbose=False):
 
     source = scrub_source(source, nsteps, verbose=True)
 
-    writefo = open(outfilename, 'w')
-    writefo.write(source)
+    with open(outfilename, 'w') as writefo:
+        writefo.write(source)
     writefo.close()
```
Comment (eric-wieser) [inline numpy/linalg/lapack_lite/clapack_scrub.py:304]: You should remove this line too

File: numpy/array_api/_typing.py [hunk 2]
```diff
@@ -37,7 +37,7 @@
 
 Device = Any
 Dtype = Type[
-    Union[[int8, int16, int32, int64, uint8, uint16, uint32, uint64, float32, float64]]
+    Union[(int8, int16, int32, int64, uint8, uint16, uint32, uint64, float32, float64)]
```
Comment (eric-wieser) [inline numpy/array_api/_typing.py:40]:
```suggestion
    Union[int8, int16, int32, int64, uint8, uint16, uint32, uint64, float32, float64]
```
is identical but more readable.

Comment (BvB93) [reply numpy/array_api/_typing.py:40]: There seems to be a bit of overlap here with https://github.com/numpy/numpy/pull/19798 though.

Comment (eric-wieser) [reply numpy/array_api/_typing.py:40]: That PR is wrong in multiple ways

Comment (sistaseetaram) [reply numpy/array_api/_typing.py]: Thank you for your suggestions... I will make these changes and commit again.

**Comment by charris** [comment]
LGTM. @BvB93 Any comments on the typing change?

**Review by charris — APPROVED** [review]

**Review by BvB93 — APPROVED** [review]
LGTM as of the latest push.

**Comment by charris** [comment]
Thanks @sistaseetaram .

**Comment by sistaseetaram** [comment]
Thank you so much.

**Pull request merged**